### PR TITLE
QA-Engine: Persist daily reports to GCS

### PR DESCRIPTION
**Pull request by alafanechere**
## What
@airbytehq/airbyte-analytics created a dedicated private bucket to store QA reports.
I want to make the QA engine write these reports to GCS in JSONL with generation in the file names.
It will be available on metabase [here](https://airbyte.metabaseapp.com/question/1810-qa-report) 

## How
* Tweak `persist_qa_report` function
* Fix some tests
* Add a CLI flag to optionally create PRs: so that we can run the daily report generation without creating PRs.

### CODE REVIEW

File: tools/ci_connector_ops/ci_connector_ops/qa_engine/outputs.py [hunk 1]
```diff
@@ -3,13 +3,18 @@
...
+from datetime import datetime
+
 import pandas as pd
 
 from .models import ConnectorQAReport
 
-def persist_qa_report(qa_report: pd.DataFrame, path: str, public_fields_only: bool =True):
+
+def persist_qa_report(qa_report: pd.DataFrame, path: str, public_fields_only: bool = True) -> str:
+    report_generation_date = datetime.strftime(qa_report["report_generation_datetime"].max(), "%Y%m%d")
```
Comment (zzztimbo) [inline tools/ci_connector_ops/ci_connector_ops/qa_engine/outputs.py:14]: Is `report_generation_datetime` different for each row in the qa_report?

Comment (alafanechere) [reply tools/ci_connector_ops/ci_connector_ops/qa_engine/outputs.py:14]: No its the same value on each row for a report.

**Review by zzztimbo — APPROVED** [review]
💯

File: .github/workflows/run-qa-engine.yml [hunk 2]
```diff
@@ -32,3 +32,4 @@ jobs:
           QA_ENGINE_AIRBYTE_DATA_PROD_SA: "${{ secrets.QA_ENGINE_AIRBYTE_DATA_PROD_SA }}"
           GITHUB_API_TOKEN: ${{ secrets.GH_PAT_MAINTENANCE_OCTAVIA }}
         run: run-qa-engine
+        # run: run-qa-engine --create-prs
```
Comment (bechurch) [inline .github/workflows/run-qa-engine.yml:35]: nit: could use a TODO or a NOTE on when to uncomment and who owns it

File: tools/ci_connector_ops/ci_connector_ops/qa_engine/main.py [hunk 3]
```diff
@@ -4,15 +4,19 @@
...
+from .constants import CLOUD_CATALOG_URL, GCS_QA_REPORT_PATH, OSS_CATALOG_URL
 
 logging.basicConfig(level=logging.INFO)
 
 logger = logging.getLogger(__name__)
 
 
-def main():
+@click.command()
+@click.option("--create-prs", is_flag=True)
```
Comment (bechurch) [inline tools/ci_connector_ops/ci_connector_ops/qa_engine/main.py:18]: Very nice

File: .github/workflows/run-qa-engine.yml [hunk 4]
```diff
@@ -2,10 +2,10 @@ name: Run QA Engine
...
 on:
   workflow_dispatch:
-  # schedule:
-  ## 1pm UTC is 6am PDT.
-  ## same time as Generate Build Report
-  # - cron: "0 13 * * *"
+  schedule:
+    # 1pm UTC is 6am PDT.
+    # same time as Generate Build Report
+    - cron: "0 13 * * *"
```
Comment (evantahler) [inline .github/workflows/run-qa-engine.yml:8]: 👍 turn it on!

File: tools/ci_connector_ops/ci_connector_ops/qa_engine/constants.py [hunk 5]
```diff
@@ -13,7 +13,7 @@
     "b76be0a6-27dc-4560-95f6-2623da0bd7b6",  # Local SQL Lite
 ]
 
-GCS_QA_REPORT_PATH = "gs://prod-airbyte-cloud-connector-metadata-service/qa_report.json"
+GCS_QA_REPORT_PATH = "gs://airbyte-data-connectors-qa-engine/"
```
Comment (evantahler) [inline tools/ci_connector_ops/ci_connector_ops/qa_engine/constants.py:16]: 👍 bucket for this purpose

Comment (evantahler) [reply tools/ci_connector_ops/ci_connector_ops/qa_engine/constants.py:16]: Which GCS project owns the bucket?

**Review by evantahler — APPROVED** [review]

**Review by bnchrch — APPROVED** [review]
Not comments from me, looks good!

File: tools/ci_connector_ops/ci_connector_ops/qa_engine/constants.py (hunk 5 above)
Comment (alafanechere) [reply tools/ci_connector_ops/ci_connector_ops/qa_engine/constants.py:16]: @zzztimbo , I think its the data analytics GCP project.

Comment (zzztimbo) [reply tools/ci_connector_ops/ci_connector_ops/qa_engine/constants.py:16]: `ab-analytics-308500`

Comment (evantahler) [reply tools/ci_connector_ops/ci_connector_ops/qa_engine/constants.py:16]: I found it! Thanks!

**Pull request merged**